src/main/java/com/google/api/codegen/transformer/OutputTransformer.java

```diff
@@ -207,8 +200,47 @@ public void run() {
     TypeModel type;
     if (baseIdentifier.equals(RESPONSE_PLACEHOLDER)) {
       view.variable(context.getNamer().getSampleResponseVarName(context));
+      boolean pageStreaming = context.getMethodConfig().getPageStreaming() != null;
+      boolean pageStreamingAndUseResourceName =
+          pageStreaming
+              && context
+                  .getFeatureConfig()
+                  .useResourceNameFormatOption(
+                      context.getMethodConfig().getPageStreaming().getResourcesFieldConfig());
+
+      // Compute the resource name format of output type and save that to local variables
+      if (pageStreamingAndUseResourceName) {
+        String typeName =
+            context
+                .getNamer()
+                .getAndSaveElementResourceTypeName(
+                    context.getTypeTable(),
+                    context.getMethodConfig().getPageStreaming().getResourcesFieldConfig());
+
+        Preconditions.checkArgument(
```

Comment:
One of the questions I had is why we're doing this only at the top level and not further down the tree (for example, somebody could set the assignment `foo = RESPONSE_PLACEHOLDER`. As we discussed offline, in that case `foo` will be in the local scope table and the checks in the loop below will catch it.

So, one thing to investigate is whether we can simplify the logic here by removing these special checks and instead putting the node corresponding to `BASE_IDENTIFIER` directly in the type table. This may also apply to non-page streaming cases, I think. (Not too sure about this, rather confusing.)

@pongad for comments

Reply:
I think resource name here is actually a special case for two reasons.

1. It's the only time we need to deal with a "synthetic type". Resource names don't have a protobuf type backing it up.

2. IIUC, resource names can only appear at the top level. GAPIC client can return resource names, or lists of resource names, but never Protobuf messages containing resource names or lists thereof. This is because resource names are known only to GAPICs, not Protobuf. When Protobuf types are generated, `protoc` doesn't know what resource names are, so response messages have no way of referring to them.